### PR TITLE
crd validation added

### PR DESCRIPTION
**Pull request by shravindran**

### CODE REVIEW

File: deploy/crd.yaml [hunk 1]
```diff
@@ -30,6 +30,40 @@ spec:
...
+                    enum:
+                      - 'admin'
+                      - 'read'
+            server:
+              type: string
+            serverType:
+              type: string
+              enum:
+                 - 'postgres'
+  subresources:
```
Comment (kumaya) [inline deploy/crd.yaml]: why did you added this ? right now we don't support CustomResourceSubResources.

File: deploy/crd.yaml [hunk 2]
```diff
@@ -30,6 +30,40 @@ spec:
...
     plural: databases
   scope: Namespaced
+  validation:
+    openAPIV3Schema:
+      properties:
+        spec:
+          required:
+            - users
+            - server
+            - serverType
```
Comment (kumaya) [inline deploy/crd.yaml]: we are expecting user to give either of `server` or `serverType`. Not sure if we should mandate both of them. Not sure if there is a better way either. :(

File: deploy/crd.yaml [hunk 3]
```diff
@@ -30,6 +30,40 @@ spec:
...
+                properties:
+                  name:
+                    type: string
+                  password:
+                    type: string
+                  role:
+                    type: string
+                    enum:
+                      - 'admin'
+                      - 'read'
```
Comment (kumaya) [inline deploy/crd.yaml]: is this verified ?

File: deploy/crd.yaml [hunk 4]
```diff
@@ -30,6 +30,40 @@ spec:
...
+            server:
+              type: string
+            serverType:
+              type: string
+              enum:
+                 - 'postgres'
```
Comment (kumaya) [inline deploy/crd.yaml]: add `mysql` as well

File: deploy/crd.yaml (hunk 3 above)
Comment (shravindran) [reply deploy/crd.yaml]: yes

File: deploy/crd.yaml (hunk 4 above)
Comment (shravindran) [reply deploy/crd.yaml]: ok

File: deploy/crd.yaml (hunk 2 above)
Comment (shravindran) [reply deploy/crd.yaml]: will make only server mandatory as server is enough to get the database plugin object.

File: deploy/crd.yaml (hunk 1 above)
Comment (shravindran) [reply deploy/crd.yaml]: will remove subresources part

File: deploy/crd.yaml [hunk 5]
```diff
@@ -30,6 +30,33 @@ spec:
     kind: Database
```
Comment (jkraj) [inline deploy/crd.yaml]: Add validation for superUser, superUserPassword, DBHost

Comment (shravindran) [reply deploy/crd.yaml]: done

**Pull request merged**